Sources/AppCoinsSDK/Models/Services/Billing/AppCoinBillingClient.swift

```diff
@@ -553,32 +553,44 @@ internal class AppCoinBillingClient : AppCoinBillingService {
         }
     }
     
-    internal func getSupportedCurrencies(result: @escaping (Result<CurrencyListRaw, BillingError>) -> Void) {
-        if let requestURL = URL(string: "\(endpoint)/currencies?type=FIAT&icon.height=128") {
-            var request = URLRequest(url: requestURL)
-            
-            request.httpMethod = "GET"
-            request.addValue("application/json", forHTTPHeaderField: "Content-Type")
-            request.timeoutInterval = 10
-            
-            let task = URLSession.shared.dataTask(with: request) { data, response, error in
+    internal func getSupportedCurrencies(result: @escaping (Result<[CurrencyRaw], BillingError>) -> Void) {
+        var allCurrencies: [CurrencyRaw] = []
+        
+        func getAllCurrencies(url: URL? = nil) {
+            if let requestURL = url ?? URL(string: "\(endpoint)/currencies?type=FIAT&icon.height=128") {
```

Comment:
When building the URL use `URLComponents` as done on the methods at the top of this class.